bundles/EcommerceFrameworkBundle/PaymentManager/Payment/PayPalSmartPaymentButton.php

```diff
@@ -164,7 +164,18 @@ protected function buildRequestBody(PriceInterface $price, array $config)
      */
     public function startPayment(OrderAgentInterface $orderAgent, PriceInterface $price, AbstractRequest $config): StartPaymentResponseInterface
     {
-        $json = $this->initPayment($price, $config->asArray());
+        $result = $this->initPayment($price, $config->asArray());
+
+        if ($result instanceof \stdClass) {
+            $json = json_encode($result);
+        } else {
+            json_decode($result);
+            if (json_last_error() == JSON_ERROR_NONE) {
+                $json = $result;
```

Comment:
hmm, am I missing something? 
right now, this if is useless ... $json always gets value of $result assigend?

Reply:
Well not always, just in case `$result` is not a `stdClass`.
The check if it is a json is in fact useless.

I was not sure, if it would make sense to throw an exception if is is not json

Reply:
changed the check and added exception

Reply:
looks better now!